src/compiler/utilities.ts

```diff
@@ -2406,4 +2406,16 @@ namespace ts {
             }
         }
     }
+
+    export function arrayStructurallyIsEqualTo<T>(array1: Array<T>, array2: Array<T>): boolean {
+        if (!array1 || !array2) {
+            return false;

```

Comment:
this is not true if both arrays are `null` / `undefined`. 
I would probably add extra parameter `sortBeforeComparison` to `arrayIsEqualTo` and sort input data inside instead of this new function - initial checks here are exactly the same with `arrayIsEqualTo` and the only difference is that inputs are sorted